tests/spec/unit/Api.spec.js

```diff
@@ -266,26 +259,37 @@ describe('Api class', () => {
             }
         ));
 
-        it('unrecognized type plugin', () => api.addPlugin({
-            id: 'unrecognized-plugin',
-            dir: pluginFixture,
-            getPlatformsArray: () => { return ['electron']; },
-            getFilesAndFrameworks: (platform) => { return []; },
-            getAssets: (platform) => {
-                return [{
-                    itemType: 'unrecognized'
-                }];
-            },
-            getJsModules: (platform) => { return []; },
-            getConfigFiles: (platform) => { return []; }
-        }, { }).then(
-            (result) => {
-                expect(result).not.toBeDefined();
-            },
-            (error) => {
-                fail('Unwanted code branch: ' + error);
-            }
-        ));
+        it('unrecognized type plugin', () => {
+            const _events = api.events;
+            const emitSpy = jasmine.createSpy('emit');
+            api.events = {
+                emit: emitSpy
+            };
+
+            api.addPlugin({
+                id: 'unrecognized-plugin',
+                dir: pluginFixture,
+                getPlatformsArray: () => { return ['electron']; },
+                getFilesAndFrameworks: (platform) => { return []; },
+                getAssets: (platform) => {
+                    return [{
+                        itemType: 'unrecognized'
+                    }];
+                },
+                getJsModules: (platform) => { return []; },
+                getConfigFiles: (platform) => { return []; }
+            }, { }).then(
+                (result) => {
+                    expect(emitSpy.calls.argsFor(0)[1]).toContain('unrecognized');
+                    expect(result).not.toBeDefined();
+                },
+                (error) => {
+                    fail('Unwanted code branch: ' + error);
+                }
+            );
+
+            api.events = _events;
+        });
```

Comment:
This is an async test. Therefore 

1. Recovering api.events should be done in finally section of Promise.
2. return Promise object as a function.

Then we would better off change that test code as
```
       it('unrecognized type plugin', () => {
          ...
         return api.addPlugin({
              id: 'unrecognized-plugin',
             ...
         ).finally(() => {
             api.events = _events;
        }); 
```
Is it right?

Reply:
I updated to return `api.addPlugin`. 

I also added the recover calls to the `success` and `fail` cb. The finally option is not supported in lower node versions.

https://node.green/#ES2018-features-Promise-prototype-finally

Reply:
It look good